lib/util/hash.spec.ts

```diff
@@ -10,9 +12,64 @@ describe('util/hash', () => {
     );
   });
 
-  test('should hash data with sha512', () => {
+  it('hashes data with sha512', () => {
     expect(hash('https://example.com/test.txt')).toBe(
       '368b1e723aecb5d17e0a69d046f8a7b9eb4e2aa2ee78e307d563c57cde45b8c3755990411aa2626c13214a8d571e0478fa9a19d03e295bb28bc453a88206b484'
     );
   });
+
+  it('correctly hashes the content of a readable stream', async () => {
+    const content = 'This is some test content.';
+    const expectedHash = crypto
+      .createHash('sha256')
+      .update(content)
+      .digest('hex');
+
+    // Create a readable stream from the content
+    const readableStream = new Readable();
+    readableStream.push(content);
+    readableStream.push(null);
+
+    const actualHash = await hashStream(readableStream, 'sha256');
+
+    expect(actualHash).toBe(expectedHash);
+  });
+
+  it('uses sha512 if no algorithm is specified', async () => {
+    const content = 'This is some test content.';
+    const expectedHash = crypto
+      .createHash('sha512')
+      .update(content)
+      .digest('hex');
+    // Create a readable stream from the content
+    const readableStream = new Readable();
+    readableStream.push(content);
+    readableStream.push(null);
+
+    const actualHash = await hashStream(readableStream);
+
+    expect(actualHash).toBe(expectedHash);
+  });
+
+  // Need help: this fails with the following msg
+  //     Node.js v18.13.0
+  // C:\Users\LENOVO\Desktop\Whitesource\renovate\lib\util\hash.spec.ts:37
+  //             const error = new Error('Stream error');
+  //                           ^
+
+  // Error: Stream error
+  // Is this what we should expect, the same happends with hasha.fromStream as well
+
+  // it('handles errors', async () => {
+  //   // Create a readable stream with an error
+  //   const readableStreamWithError = new Readable();
+  //   const error = new Error('Stream error');
+  //   readableStreamWithError._read = () => {
+  //     readableStreamWithError.emit('error', error);
+  //   };
+
+  //   expect(
+  //     async () => await hashStream(readableStreamWithError, 'sha256')
+  //   ).toThrow();
+  // });
```

Comment:
I wanted to catch and log, when an error occurs while reading from the stream, but couldn't do it. 
This test case reproduces such an error. 

Instead of error being caught the execution stops with Stream Error message. So if today while reading from stream with `hasha.fromStream` an error occurs I think the repo execution will halt.

Reply:
You should remove that comment or enable the test

Reply:
I will remove it since that is the current apporach. I have a diubt though, is it okay to not use a catch block here?